lib/client-assets.php

```diff
@@ -306,6 +306,29 @@ function gutenberg_scripts_and_styles( $hook ) {
 	 */
 	wp_enqueue_media();
 
+	// Export REST bases for all post types.
+	$post_type_rest_base_mapping = array();
+	foreach ( get_post_types( array(), 'objects' ) as $post_type_object ) {
+		$rest_base = ! empty( $post_type_object->rest_base ) ? $post_type_object->rest_base : $post_type_object->name;
+		$post_type_rest_base_mapping[ $post_type_object->name ] = $rest_base;
+	}
+	$script = sprintf( 'wp.api.postTypeRestBaseMapping = %s;', wp_json_encode( $post_type_rest_base_mapping ) );
+	$script .= <<<JS
```

Comment:
Just thinking that this function is doing a lot of work, when it should be simple enqueues. Are these overrides specific to the editor? Maybe binding to anywhere `wp-api` is enqueued. Or at least a separate function / file, or even separate action hook? Similarly below with retrieving and validating post to be edited.